### DISCUSSION TITLE
Salinan Data Transaksi Sebagai Backup ketika Sistem Down

### DISCUSSION THREAD

**Discussion by nafiesl** [buku-masjid/buku-masjid#46] [Ideas]
### Problem

Ada kasus di mana sebuah aplikasi buku masjid (self-hosted) down, ini berlangsung lebih dari 1 hari. Pengurus masjid jadi tidak bisa meng-input data transaksi karena sistem tidak bisa diakses. Jika pengurus masjid (bendahara) sudah bergantung dengan sistem ini (misal tidak ada salinan riwayat transaksi di tempat lain), maka ini jadi masalah besar.

### Potensi dampak

Dengan kondisi ini, ada potensi **riwayat transaksi hilang** jika database tidak sempat ter-backup.

### Ide pencegahan

Ada sebuah ide untuk membantu pengurus masjid melakukan export data perbulan, seperti download mutasi rekening di internet banking. Idenya kurang lebih seperti ini:

1. Di sebuah halaman admin (entah di bagian mana), ada tombol untuk download excel
2. Format excel yang ter-download mungkin seperti tampilan laporan rincian transaksi (screenshot di bawah)
3. File excel dapat di-download per bulan, atau rentang tanggal (mana yang lebih mudah)

![screen_2024-02-23_009](https://github.com/buku-masjid/buku-masjid/assets/8721551/013ff501-d947-4903-9c74-52416ffd9979)

### Benefit yang diharapkan

1. Dengan diberikan fasilitas download file excel ini, diharapkan agar file excel dapat dijadikan sebagai backup/salinan untuk pegangan bendahara atas data transaksi yang sudah diinput ke sistem.
2. Ketika sistem down, bendahara dapat mencatat transaksi dulu pada file excel (sembari menunggu sistem up lagi).
3. Ketika sistem sudah up lagi, bendahara bisa menginput transaksi yang dicatat manual di excel tadi.

Sebenarnya bendahara bisa saja mencatat transaksi di file excel dengan format dibuatnya sendiri, tetapi jika file excel nya sudah disediakan oleh sistem bukumasjid, diharapkan pencatatan transaksi "sementara" akan lebih mudah bagi bendahara.

Gimana menurut teman-teman? Mungkin ada ide lain. Terima kasih

**Comment by thirthfamous** [comment]
hmmm kalau memang di servernya tidak ada rutinitas backup (ke s3 misalkan) paling ide saya skrg bisa sediakan fitur backup seluruh data untuk user, di download sama user.

**Reply by nafiesl** [reply]
Untuk saat ini, backup data sudah disediakan mas. Ini untuk backup dan restore database manual.

![Screenshot_20240224-195950_Firefox](https://github.com/buku-masjid/buku-masjid/assets/8721551/400e17f3-0d74-49d7-b8aa-0a28714ee644)
